DataBiosphere/terra-workspace-manager
# [WOR-1588] Spring Boot 3.2.3, k8s client 20.0.1, TCL 1.1.6, LZS 0.0.317

Notable upgrades:
- Spring Boot 3.1.2 -> 3.2.3
- TCL 0.1.9 -> 1.1.6
- LZS 0.0.305 -> 0.0.317
- k8s client 18.0.0 -> 20.0.1

Collectively, these upgrades enable the following:
- https://broadworkbench.atlassian.net/browse/WOR-1588 - Upgrading k8s client past breaking changes in 20.0.0
- https://broadworkbench.atlassian.net/browse/WOR-1566 - Addressing security vulnerabilities in transitive dependencies
- https://broadworkbench.atlassian.net/browse/WOR-1560 - Taking advantage of Stairway 1.x's native MDC awareness
- https://broadworkbench.atlassian.net/browse/WOR-1590 - Pull in LZS fix

While doing this work:
- Dependency clean-up and light reorganization (but there is yet more work to be done here)
- Tell Dependabot to try to update Spring Boot, and to stop trying to update k8s client due to their intermingling of legacy and non-legacy versions
- Wired in srcclr plugins and added instructions for running scans off of local code and feature branches

SCA scan run off this feature branch: https://github.com/broadinstitute/dsp-appsec-sourceclear-github-actions/actions/runs/8513427070/job/23317060997

**Manual Verification**

I first deployed this branch to my BEE successfully… but WSM BEEs don't use pub-sub queues for their Stairway flights, and the original impetus for rolling back WSM's k8s client upgrade was a runtime issue with TCL's KubePodListener that only manifests in live (non-BEE) environments.

I then [pointed my BEE](https://app.gitbook.com/o/-LVj1clrBdFW94pG2z2w/s/-M1f_HVXo41b0qPSpsGF/kubernetes/so-you-want-to-deploy-your-service-to-perf#appendix-a) to a [terra-helmfile draft PR](https://github.com/broadinstitute/terra-helmfile/pull/5352) which sets all deployments to run `inKubernetes` (use pub-sub queues for Stairway flights).  But my WSM [failed to start up](https://cloudlogging.app.goo.gl/NJKcBhzAFybEEZkf9) due to missing a permission to create the topic:
```
"Application run failed\ncom.google.api.gax.rpc.PermissionDeniedException: io.grpc.StatusRuntimeException: PERMISSION_DENIED: User not authorized to perform this action.\n\t
...
at bio.terra.stairway.gcp.GcpQueueUtils.makeTopic(GcpQueueUtils.java:56)
```

I asked [DevOps for help](https://broadinstitute.slack.com/archives/CADM7MZ35/p1712011058792129?thread_ts=1707152808.522839&cid=CADM7MZ35) in a recent related thread.  Chelsea temporarily granted the WSM service account `pubsub.editor` in the `broad-dsde-qa` project, and I was able to successfully verify `KubePodListener` behavior on my BEE: https://cloudlogging.app.goo.gl/9m6XfG8QAPnXmbZZ7
